File: 2025/day02/part01.py
with open("input.txt") as f:
    content = f.read()


def garbage(n: int):
    s = str(n)
    l = len(s)
    if l % 2 != 0:
        return False
    s1 = s[: l // 2]
    s2 = s[l // 2 :]
    return s1 == s2


content = [elem.split("-") for elem in content.strip().split(",")]
c = 0
for r in content:
    a = int(r[0])
    b = int(r[1])
    for i in range(a, b + 1):
        if garbage(i):
            print(i)
            c += i

print(c)

File: 2025/day05/Part01.hs
module Part01 where

import Text.Read (readMaybe)

parseHeader :: String -> Maybe (Int, Int)
parseHeader xs =
  case break (== '-') xs of
    (f, '-' : s) -> (,) <$> readMaybe f <*> readMaybe s
    _ -> Nothing

isFresh :: [(Int, Int)] -> Int -> Bool
isFresh xs n = any (\(l, u) -> l <= n && n <= u) xs

main :: IO ()
main = do
  content <- lines <$> readFile "input.txt"
  let (header, "" : footer) = break (== "") content
  let Just header' = mapM parseHeader header
  let Just (footer' :: [Int]) = mapM readMaybe footer
  let isFresh' = isFresh header'
  print $ length $ filter id $ map isFresh' footer'
  return ()
